File: VocaDbModel/DataContracts/UseCases/CreateSongContract.cs
#nullable disable

using VocaDb.Model.DataContracts.Songs;
using VocaDb.Model.Domain.Songs;

namespace VocaDb.Model.DataContracts.UseCases
{
	public class CreateSongContract
	{
		public ArtistForSongContract[] Artists { get; init; }

		public bool Draft { get; init; }

		public LyricsForSongContract[] Lyrics { get; init; }

		public LocalizedStringContract[] Names { get; init; }

		public SongContract OriginalVersion { get; init; }

		public string[] PVUrls { get; set; }

		public string ReprintPVUrl { get; init; }

		public SongType SongType { get; set; }

		public string UpdateNotes { get; init; }

		public WebLinkContract[] WebLinks { get; init; }
	}
}

File: VocaDbWeb/Scripts/DataContracts/Song/CreateSongContract.ts
import LocalizedStringContract from '@/DataContracts/Globalization/LocalizedStringContract';
import ArtistForSongContract from '@/DataContracts/Song/ArtistForSongContract';
import SongContract from '@/DataContracts/Song/SongContract';
import SongType from '@/Models/Songs/SongType';

// Corresponds to the CreateSongForApiContract record class in C#.
export default interface CreateSongContract {
	artists: ArtistForSongContract[];
	draft: boolean;
	names: LocalizedStringContract[];
	originalVersion?: SongContract;
	pvUrls: string[];
	reprintPVUrl: string;
	songType: SongType;
}
